dags/common/utils.py

```diff
@@ -59,3 +61,17 @@ def parse_to_int(value):
         return int(value)
     except (ValueError, TypeError):
         logger.error("Cannot parse to integer", value=value)
+
+
+def preserve_cdata(article: str):
+    matches = CDATA_PATTERN.finditer(article)
+    for match in matches:
+        cdata_content_with_escaped_escape_chars = match.group(1).replace("\\", "\\\\")
+        article = CDATA_PATTERN.sub(
+            cdata_content_with_escaped_escape_chars, article, count=1
+        )
+    return article
```

Comment:
regex are quite expensive, are sure we want to run this in everything? Also, let's add unit tests for this specific function.

Reply:
If I thinking better, maybe not. CDATA usually has mathematical expressions which can be in title/s or abstract. So, maybe it would be enough to run just on them. I will add tests

Reply:
Yes please just run it for the ones that we really need.

Reply:
We can see in the tests that just by preserving the CDATA by using regex, we can get the value inside of it: 
https://github.com/cern-sis/workflows/blob/3f3239709d827cf0ef558cf56d485a592344a08f/tests/units/iop/test_iop_parser.py#L362

If we are not doing it before parsing to ET element, we lose the CDATA:
https://github.com/cern-sis/workflows/blob/3f3239709d827cf0ef558cf56d485a592344a08f/tests/units/iop/test_iop_parser.py#L390

Even when we are parsing it back to a string:
https://github.com/cern-sis/workflows/blob/3f3239709d827cf0ef558cf56d485a592344a08f/tests/units/iop/test_iop_parser.py#L397

Reply:
You can use [this function](https://github.com/inspirehep/inspire-utils/blob/6eb48fbf93963b4a5809318cf1fbc19f58db602b/inspire_utils/helpers.py#L113)

Reply:
I am doing it already:
https://github.com/cern-sis/workflows/blob/704383598780d0f667475a80a87abdc63c98db44/dags/common/parsing/xml_extractors.py#L32|

Is not preserving CDATA, so I need a regex in any case

Reply:
yes, but this is just an example that the XML is not getting back to the same structure as it was before parsing it to ElemenTree. In the article parsing, I am not sure do I need it, because I am taking the element until the deepest level, it means I have: inline-graphic, tex-math and etc. These should be kept, as far as I understand?( @drjova )

Reply:
Should I still remove tags?